cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramRunStatusMonitorService.java

```diff
@@ -170,4 +181,30 @@ private Predicate<RunRecordDetail> createFilter(long currentTimeInSecs, Set<Prog
       return terminateTime != null && terminateTime <= currentTimeInSecs;
     };
   }
+
+  private void emitForceTerminatedRunsMetric(ProgramRunId programRunId, RunRecordDetail recordedRunRecord) {
+    Optional<ProfileId> profile = SystemArguments.getProfileIdFromArgs(programRunId.getNamespaceId(),
+                                                                       recordedRunRecord.getSystemArgs());
+    Map<String, String> additionalTags = new HashMap<>();
+    // don't want to add the tag if it is not present otherwise it will result in NPE
+    additionalTags.computeIfAbsent(Constants.Metrics.Tag.PROVISIONER,
+                                   provisioner -> SystemArguments.getProfileProvisioner(
+                                     recordedRunRecord.getSystemArgs()));
+    profile.ifPresent(profileId -> {
+      emitProfileMetrics(programRunId, profileId, additionalTags);
+    });
+  }
+
+  private void emitProfileMetrics(ProgramRunId programRunId, ProfileId profileId,
+                                  Map<String, String> additionalTags) {
+    Map<String, String> tags = ImmutableMap.<String, String>builder()
+      .put(Constants.Metrics.Tag.PROFILE_SCOPE, profileId.getScope().name())
+      .put(Constants.Metrics.Tag.PROFILE, profileId.getProfile())
+      .put(Constants.Metrics.Tag.NAMESPACE, programRunId.getNamespace())
+      .put(Constants.Metrics.Tag.PROGRAM, programRunId.getProgram())
+      .putAll(additionalTags)
+      .build();
+    metricsCollectionService.getContext(tags).increment(Constants.Metrics.Program.PROGRAM_FORCE_TERMINATED_RUNS, 1L);
```

Comment:
should this also be using `ProgramRunners.createProgramMetricsContext`? For example, as currently written, the app will not be in the tags

Reply:
As discussed offline, keeping the creation of metric context the same as it also adds runId as a tag.
Instead added the App tag to the current set of tags